core/auth.py

```
@@ -0,0 +1,356 @@
+                        'metadata': key.metadata
+                    }
+                    for key in self.api_keys.values()
+                ]
+            }
+
+            with open(self.keys_file, 'w') as f:
+                json.dump(data, f, indent=2)
+
+            logger.info(f"Saved {len(self.api_keys)} API keys to {self.keys_file}")
+
+        except Exception as e:
```

Ruff (0.14.10)

Do not catch blind exception: `Exception`

(BLE001)